language/bytecode-verifier/src/type_memory_safety.rs

```diff
@@ -460,7 +460,7 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
                             Err(err_at_offset(StatusCode::COPYLOC_RESOURCE_ERROR, offset))
                         }
                         Kind::Unrestricted => {
-                            if !state.is_local_borrowed(*idx) {
+                            if !state.is_local_mutably_borrowed(*idx) {
```

Comment:
Minor suggestion:

pub fn is_local_immutably_borrowed(&self, idx: LocalIndex) -> bool {
        self.all_nonces_immutable(self.local_borrows(idx))
}

Use the above function both at line 463 and to implement 

pub fn is_local_mutably_borrowed(&self, idx: LocalIndex) -> bool {
    !self.is_local_immutably_borrowed(idx)
}

Otherwise, I was scratching my head a bit to understand the double negation, one at line 463 and one inside the implementation.

Reply:
I think that would be a bit misleading, because `is_local_mutably_borrowed` returns false even if there are no borrows at all

Having `is_local_immutably_borrowed` return true when it is not borrowed would be weird to me